integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/NoPagingPMT.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.it.panache;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.h2.tools.Server;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.builder.Version;
+import io.quarkus.test.LogFile;
+import io.quarkus.test.QuarkusProdModeTest;
+import io.restassured.RestAssured;
+
+/**
+ * Test if PanacheQuery is using unnecessary SQL for limiting the number of output rows, the log output is tested for the
+ * presence of <code>offset</code> or <code>limit</code> in the SQL.
+ */
+public class NoPagingPMT {
+
+    static {
+        // WORKAROUND
+        try {
+            Server.createTcpServer().start();
```

Comment:
This worries me a bit, why is it needed and where is it stopped?

Reply:
As ProdModeTest build a small JAR and run the test on a different JVM, There is a need to start the H2 Database server, My question to @geoand that I will detail later is about this, The workaround to start it manually is because the ProdModeTest isn't starting it even when I am adding `TestResources` that has the `@QuarkusTestResource(H2DatabaseTestResource.class)` annotation

As a temporary workaround didn't closed it myself because it is closed when that specific test JVM is closed.

Reply:
Yeah, the prod mode test doesn't handle resources...

Reply:
We could make it handle them for sure, but currently it doesn't.